### PR TITLE
[BUG] CondIsPreferredTool invalid checks

### PR DESCRIPTION
**Pull request by Fusezion**
### Description
The great cheese lord brought to my attention that `CondIsPreferredTool` contained a flaw where in certine edge cases it could return true when intended to be false, as such I've kindly showed my loving side after being appointed by Kenzie on discord to fix this bug. 

Based off my original test this was caused only when block was null and didn't matter for items
new check method was based off of how we do CondItemInHand

**Old Behavior**
image

**New Behavior** *still returns true for negated*
image

---
**Target Minecraft Versions:** any
**Requirements:** N/A
**Related Issues:** #5397

### CODE REVIEW

**Review by AyhamAl-Ali — APPROVED** [review]
Good Job ⚡

File: src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java [hunk 1]
```diff
@@ -78,20 +78,17 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
...
+				item -> {
+					if (block instanceof Block) {
+						return ((Block) block).isPreferredTool(item.getRandom());
+					} else if (block instanceof BlockData) {
+						return ((BlockData) block).isPreferredTool(item.getRandom());
+					} else {
+						return false;
+					}
+				}), isNegated());
 	}
```
Comment (Fusezion) [inline src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java:92]: Say I changed `return false;` to `return isNegated();` that should fail for if both are null yes?

Comment (DelayedGaming) [reply src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java:92]: The latter is expected to return `true` due to negation

Comment (Fusezion) [reply src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java:92]: yeah didn't matter anyways when I tried test couldn't even run, for now just gonna fix the fact block data test don't actually use block data

**Comment by Fusezion** [comment]
Alrighty thanks for coming to my ted talk I'm heading to sleep before I hear more geese

**Review by AyhamAl-Ali — APPROVED** [review]

**Review by DelayedGaming — APPROVED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java [hunk 2]
```diff
@@ -78,20 +78,17 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
...
-				// invalid type
-				return false;
-			}
-		}
-		// all checks passed
-		return true;
+		return blocks.check(event,
+			block -> items.check(event,
+				item -> {
+					if (block instanceof Block) {
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/conditions/CondIsPreferredTool.java:84]:
```
		return blocks.check(event, block ->
				items.check(event, item -> {
					if (block instanceof Block) {
```

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

**Review by TheLimeGlass — APPROVED** [review]

**Pull request merged**